lib/Gateway.js

```diff
@@ -786,6 +807,77 @@ function addRgbDimmer (cfg, node, valueId) {
   }
 
   return false
+} */
+
+function addRgbColorSwitch (cfg, node, valueId) {
+  cfg = copy(cfg)
```

Comment:
you are coping a switch config, you need to change the discovery payload to match this to light entity instead